packages/cli/lib/deamon.js

```diff
@@ -0,0 +1,195 @@
+const urllib = require('urllib');
+const AutoLaunch = require('auto-launch');
+const path = require('node:path');
+const fs = require('node:fs/promises');
+const { chmodSync } = require('node:fs');
+const { rsBindingPath } = require('@cnpmjs/binding');
+const execa = require('execa');
+
+const {
+  baseRapidModeDir,
+  nydusd,
+  nydusdConfigFile,
+  nydusdMnt,
+  socketPath,
+  nydusdLogFile,
+} = require('./constants');
+
+const deamonDir = path.join(baseRapidModeDir(), 'project');
+
+const metadataDir = path.join(deamonDir, 'metadata');
+
+const rapidDeamon = rsBindingPath
+  ? path.join(rsBindingPath, 'rapid_deamon')
+  : undefined;
+
+
+const destinationFilePath = path.join(deamonDir, 'rapid_deamon');
+
+const daemonPoint = 'http://localhost:33889';
```

Comment:
For local daemon processes, we generally use Unix Domain Sockets (UDS) for communication.